agent/agent/src/main/java/com/microsoft/applicationinsights/agent/StartupProfiler.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * ApplicationInsights-Java
+ * Copyright (c) Microsoft Corporation
+ * All rights reserved.
+ *
+ * MIT License
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this
+ * software and associated documentation files (the ""Software""), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+ * persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE
+ * FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+ * OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package com.microsoft.applicationinsights.agent;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+import java.lang.management.ThreadInfo;
+import java.lang.management.ThreadMXBean;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+final class StartupProfiler {
+
+  private static final String STACKTRACES = "stacktrace.txt";
+
+  @SuppressWarnings("SystemOut")
+  public static void start() {
+    // this is used to support -Dapplicationinsights.debug.startupProfiling=true
+    if (!Boolean.parseBoolean(System.getProperty("applicationinsights.debug.startupProfiling"))) {
+      return;
+    }
+
+    if (isReadonly()) {
+      System.out.println("It's a readonly file system. StartupProfiler does nothing in this case.");
+      return;
+    }
+
```

Comment:
no need for nice behavior on read-only system for this feature, good enough error will get logged when trying to write to the file

```suggestion
```